src/PersonObjects/Sleeve/Work/SleeveBladeburnerWork.ts

```diff
@@ -34,9 +34,26 @@ export class SleeveBladeburnerWork extends Work {
   process(player: IPlayer, sleeve: Sleeve, cycles: number): number {
     if (!player.bladeburner) throw new Error("sleeve doing blade work without being a member");
     this.cyclesWorked += cycles;
+    const actionIdent = player.bladeburner.getActionIdFromTypeAndName(this.actionType, this.actionName);
+    if (!actionIdent) throw new Error(`Error getting ${this.actionName} action`);
+    if (this.actionType === "Contracts") {
+      const action = player.bladeburner.getActionObject(actionIdent);
+      if (!action) throw new Error(`Error getting ${this.actionName} action object`);
+      if (action.count <= 0) {
+        sleeve.stopWork(player);
+        return 0;
+      }
+    }
+
     while (this.cyclesWorked > this.cyclesNeeded(player, sleeve)) {
-      const actionIdent = player.bladeburner.getActionIdFromTypeAndName(this.actionType, this.actionName);
-      if (!actionIdent) throw new Error(`Error getting ${this.actionName} action`);
+      if (this.actionType === "Contracts") {
+        const action = player.bladeburner.getActionObject(actionIdent);
+        if (!action) throw new Error(`Error getting ${this.actionName} action object`);
+        if (action.count <= 0) {
+          sleeve.stopWork(player);
+          return 0;
+        }
+      }
```

Comment:
i dont think we need to duplicate this code just having it once in the while loop should do the same thing,

or is there some condition that makes it keep working and the while loop is never entered?

Reply:
actually looks like youve covered this code in your other pr mind removing for this one?

Reply:
@phyzical 
> i dont think we need to duplicate this code just having it once in the while loop should do the same thing,

I put it there twice because the first check would only happen during the action and second happens after the action. if only the first is there one could get tons of cycles and still go to negative contracts. if only the second is there the game would take to finish the action and then not give any rewards, It's not much time tho. Removing the first check and adding a check to when the action is started would be better, but I couldn't figure out how to add that

> or is there some condition that makes it keep working and the while loop is never entered?

no

> actually looks like youve covered this code in your other pr mind removing for this one?

I'm actually not sure how this commit ended up on this branch -.-'